docs/admin/master-node-communication.md

```diff
@@ -83,3 +83,57 @@ cluster (connecting to the ssh server listening on port 22) and passes all
+
+Workflow 
+
+##### apiserver configuration
+A token file specifies at least one token as a "bootstrap token" assigned to a kubelet boostrap-specific group. This group will later be used in the controller-manager configuration to scope approvals in the default approval controller. As this feature matures, this token should also be bound to an RBAC policy that limits requests using the bootstrap token to only be able to make requests related to certificate provisioning. When RBAC policy is in place, scoping the tokens to a group will allow great flexibility (e.g. you could disable a particular bootstrap group's access when you are done provisioning the nodes). 
+
+##### Token auth file 
+Tokens are arbitrary, but should represent at least 128 bits of entropy derived from a secure random number generator (such as /dev/urandom on most modern systems). There are lots of ways you could generate tokens; one is: 
+
+`head -c 16 /dev/urandom | od -An -t x | tr -d ' '` 
+
+which will generate tokens that look like `02b50b05283e98dd0fd71db496ef01e8` 
+
+The token file will look like this, where the first three values can be anything and the quoted group name should be as depicted: 
+
+```
+02b50b05283e98dd0fd71db496ef01e8,kubelet-bootstrap,10001,"system:kubelet-bootstrap"
+``` 
+
+Add the `--token-auth-file=FILENAME` flag to the apiserver command to enable the token file. See docs at http://kubernetes.io/docs/admin/authentication/#static-token-file for further details 
+
+#### controller-manager configuration
+The new feature adds a certificate-issuing control loop to the KCM. This takes the form of a [cfssl](https://blog.cloudflare.com/introducing-cfssl/) local signer using assets on disk. Currently, all certificates issued have one year validity and a default set of key usages. 
+
+##### Signing assets 
+These provide the signer with the cryptographic materials necessary to issue certificates. This CA should be trusted by the apiserver for authentication with the `--client-ca-file=SOMEFILE` flag. The management of the CA is beyond the scope of this document but it is recommended that you generate a dedicated CA for Kubernetes. Both certificate and key are assumed to be PEM-encoded. 
+
+The new controller-manager flags are: 
+```
+--cluster-signing-cert-file="/etc/path/to/kubernetes/ca/ca.crt" --cluster-signing-key-file="/etc/path/to/kubernetes/ca/ca.key"
+``` 
+
+##### Auto-approval 
+To ease deployment and testing, the alpha version of the feature includes a flag to approve all certificate requests made by users in a certain group. The intended use of this is to whitelist only the group corresponding to the bootstrap token in the token file above. Use of this flag circumvents makes the "approval" process described below and is not recommended for production use. 
```

Comment:
Refer to feature by name. That makes it so that the feature name and the capability are more closely coupled for search.